### ISSUE TITLE
Using a tilde in the URL of a Markdown image tag breaks the tag if the image is in the IMAGE_PATH

### ISSUE THREAD

**Issue by azrogers** [doxygen/doxygen#11273]
**Describe the bug**
Using a Markdown image tag with a `~` character in the image URL (like `![](Example~/test.png)` to an image in the `IMAGE_PATH`, the following output is produced:
```html
<img src="Example" alt=""/>
<div class="caption">
    ~/test.png
</div>
```
This, of course, breaks the image, not to mention that having path of the image path in the caption is rather undesirable. The easy workaround would be to rename the folder containing images to something that doesn't contain a tilde, but we are using Doxygen to produce documentation for a Unity package, and [adding a tilde to the end of the folder name](https://docs.unity3d.com/Manual/SpecialFolders.html) allows the folder to be ignored by Unity. Otherwise, our documentation folder shows up in the Unity assets folder, which isn't intended.

This only seems to happen when the image is in the `IMAGE_PATH`.

**Screenshots**
![image](https://github.com/user-attachments/assets/888e4b1a-5f91-4c42-b137-c57db97a0bb9)

**To Reproduce**
Here is a bare-minimum example of this issue: [doxygen-bug.zip](https://github.com/user-attachments/files/18131736/doxygen-bug.zip)

**Expected behavior**
The intended output should be:
```html
<img src="test.png" alt="" />
```

**Version**
Doxygen `1.12.0 (c73f5d30f9e8b1df5ba15a1d064ff2067cbb8267)` running on Windows 11 Pro 64-bit. The same result also occurs when running the same version of Doxygen under WSL2 running Ubuntu 24.04.1 LTS using a binary obtained from the GitHub release.

**Comment by albert-github** [comment]
Special characters in file / directory names are always a nuisance and especially when a specific package assigns a special meaning to it. (I'm not talking about the special meaning that e.g. *nix gives to a starting `~` or a starting `.` as these are not package specific but related to the OS).

The mentioned line
```
![](Example~/test.png)
```
translates internally in doxygen to:
```
@image html Example~/test.png \ilinebr @image latex Example~/test.png \ilinebr @image rtf Example~/test.png \ilinebr @image docbook Example~/test.png \ilinebr @image xml Example~/test.png \ilinebr
```


It looks like that in the file / directory parsing the `~` is not considered (I quickly check doctokenizer, commentscan and scanner).

@doxygen, should the `~ be added as (non first) character with a special meaning?

**Comment by doxygen** [comment]
@albert-github the `~` as the first character indeed has a special meaning in Unix shells, but this is only if it appears as the first character, so I think we should indeed allow `~` as the non-first character of a file or directory name.

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #11294

**Comment by albert-github** [comment]
Code has been integrated in master on GitHub (please don't close the issue as this will be done at the moment of an official release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.13.0.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
